pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -290,9 +290,7 @@ type WorkflowSpec struct {
 	// VolumeClaimTemplates is a list of claims that containers are allowed to reference.
 	// The Workflow controller will create the claims at the beginning of the workflow
 	// and delete the claims upon completion of the workflow
-	// +patchStrategy=merge
-	// +patchMergeKey=name
-	VolumeClaimTemplates []apiv1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,6,opt,name=volumeClaimTemplates"`
+	VolumeClaimTemplates []apiv1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty" protobuf:"bytes,6,opt,name=volumeClaimTemplates"`
```

Comment:
This follows the same convention as `Volumes` above.

Reply:
@terrytangyuan 

Hello. Thank you for the review! :)
If `VolumeClaimTemplates` are supposed to follow the same convention as `Volume`, can we proceed by removing `patchStrategy` and `patchMergeKey` from the Volume structure?
I believe that the approach of using `patchMergeKey` for `name` within `VolumeClaimTemplates` is currently not feasible.

Using `patchMergeKey` to reference a nested field structure in Kubernetes is not natively supported. `patchMergeKey` can only be applied to one of the top-level fields of an object, and it cannot be applied to nested fields.

To achieve the desired outcome of using name as `patchMergeKey`, you might need to consider altering the structure of the `PersistentVolumeClaim` type within `VolumeClaimTemplates`. This could involve actions such as moving the name field within the `VolumeClaimTemplates` array.

However, be cautious since making such structural changes to accommodate this might introduce unintended consequences. Modifying the structure of the structure might lead to other side effects, so it's essential to approach this with care.

Reply:
This might be a potential breaking change so I'd love to see more 👀 on this from others